test/test_sample.py

```diff
@@ -1795,6 +1795,71 @@ def test_attrs(self):
         with self.assertRaisesRegex(AttributeError, 'Unknown variable name:'):
             dummy = fit.c
 
+    # pylint: disable=use-a-generator
+    def test_diagnostics(self):
+        # centered 8 schools hits funnel
+        stan = os.path.join(DATAFILES_PATH, 'eight_schools.stan')
+        model = CmdStanModel(stan_file=stan)
+        rdata = os.path.join(DATAFILES_PATH, 'eight_schools.data.R')
+        with LogCapture() as log:
+            logging.getLogger()
+            fit = model.sample(
+                data=rdata,
+                seed=55157,
+                show_progress=False,
+                show_console=False,
+            )
+
+            self.assertTrue(
+                all([a == b for a, b in zip(fit.max_treedepths, [0, 0, 0, 6])])
+            )
+            self.assertTrue(
+                all([a == b for a, b in zip(fit.divergences, [10, 143, 5, 4])])
+            )
+
+            log.check_present(
+                (
+                    'cmdstanpy',
+                    'WARNING',
+                    'Some chains may have failed to converge.\n'
```

Comment:
Similarly, maybe we should just test that this is output, and match the rest of the message with a wildcard.

Reply:
you'd think wildcard was possible, but that's not how `log.check` and `log.check_present` work.
found a workaround.

Reply:
In other places we’ve used the StringComparison class from the same test utilities, I think that would be less brittle if another warning ever pops up

Reply:
thanks - much better!